pkg/test/runner/host/host.go

```diff
@@ -129,35 +115,60 @@ func (r *hostRunner) Run(ctx context.Context, testID string, testIndex int, test
 	return nil
 }
 
-// delegateToContainer delegates the execution of the test to a container, created and tuned as per test specification.
-func (r *hostRunner) delegateToContainer(ctx context.Context, testID string, testIndex int,
+// setUpContext sets up the context specified in the provided test description.
+func (r *hostRunner) setUpContext(ctx context.Context, logger logr.Logger, testID string,
 	testDesc *loader.Test) error {
-	// Initialize labels for the container's process.
-	labels := r.Labels
-	if labels == nil {
-		labels = &label.Set{TestIndex: testIndex, ProcIndex: 0, IsContainer: true}
-	} else {
-		labels.ProcIndex++
+	testContext := testDesc.Context
```

Comment:
Should we check that the `testDesc` is not nil?

Reply:
TestDesc validity (so non-nil-ness) is enforced at load time. I would avoid re-check it here.

Reply:
:+1: